lib/compat/wordpress-6.2/class-wp-theme-json-6-2.php

```diff
@@ -216,4 +235,45 @@ class WP_Theme_JSON_6_2 extends WP_Theme_JSON_6_1 {
 			'textTransform'  => null,
 		),
 	);
+
+	/**
+	 * Processes a style node and returns the same node
+	 * without the insecure styles.
+	 *
+	 * @since 5.9.0
+	 * @since 6.2.0 Allow inferred properties used outside of `compute_style_properties`.
+	 *
+	 * @param array $input Node to process.
+	 * @return array
+	 */
+	protected static function remove_insecure_styles( $input ) {
+		$output       = array();
+		$declarations = static::compute_style_properties( $input );
+
+		foreach ( $declarations as $declaration ) {
+			if ( static::is_safe_css_declaration( $declaration['name'], $declaration['value'] ) ) {
+				$path = static::PROPERTIES_METADATA[ $declaration['name'] ];
+
+				// Check the value isn't an array before adding so as to not
+				// double up shorthand and longhand styles.
+				$value = _wp_array_get( $input, $path, array() );
+				if ( ! is_array( $value ) ) {
+					_wp_array_set( $output, $path, $value );
+				}
+			}
+		}
+
+		foreach ( static::INFERRED_PROPERTIES_METADATA as $property => $path ) {
+			$value = _wp_array_get( $input, $path, array() );
+			if (
+				isset( $value ) &&
+				! is_array( $value ) &&
+				static::is_safe_css_declaration( $property, $value )
+			) {
+				_wp_array_set( $output, $path, $value );
+			}
+		}
```

Comment:
Is there any reason why this shouldn't be done in `remove_insecure_properties` instead? I see that method has a similar logic to [bring back pseudo selectors that are removed by `remove_insecure_styles`](https://github.com/WordPress/gutenberg/blob/af8bf6e305dc8f2478c8991b8eb2dd92b4b6456e/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L316-L325), so maybe it makes sense to have the new logic there too for consistency?

I guess the implementation would be slightly harder that way. And maybe this is the right place since this logic is strictly related to how `compute_style_properties` behaves.

Feel free to disregard this question if it's silly 😅

Reply:
That makes sense, thanks!